### PR TITLE
Move the CI to 4.12 multicore and github actions

### PR DESCRIPTION
**Pull request by ctk21**
This PR moves the CI from Travis to Github actions. 

It also updates the version of multicore used in the CI to `4.12+domains+effects`.

I had trouble getting our script to work with macOS, but I'm a bit unclear why running the CI for domainslib against macOS is valuable.

### CODE REVIEW

**Comment by ctk21** [comment]
I updated the PR to make use of the [setup-ocaml github action](https://github.com/avsm/setup-ocaml/)

Things work on macOS now. It also allows us to drop all the boilerplate for installing opam in this repo.

**Review by Sudha247 — COMMENTED** [review]
Thanks! Great to have the CI updated to 4.12. This looks good to me.

File: .ci.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+
+# use opam to install multicore and dune
+opam update
+opam switch create ${OCAML_VERSION} --packages=ocaml-variants.${OCAML_VERSION} --repositories=multicore=git+https://github.com/ocaml-multicore/multicore-opam.git,default
+eval $(opam config env)
```
Comment (avsm) [inline .ci.sh:5]:
```suggestion
eval $(opam env)
```

File: .ci.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+
+# use opam to install multicore and dune
+opam update
```
Comment (avsm) [inline .ci.sh:3]:
```suggestion
opam update -u
```

**Review by avsm — APPROVED** [review]
just minor changes, lgtm

**Pull request merged**